transform.ml

```diff
@@ -2,16 +2,17 @@ open Instr
 
 let remove_empty_jmp prog =
   let pred = Analysis.predecessors prog in
+  let succ = Analysis.successors prog in
   let rec remove_empty_jmp pc =
     let pc' = pc + 1 in
     if pc' = Array.length prog then [prog.(pc)] else
       match (prog.(pc), prog.(pc')) with
       | (Goto l1, Label l2) when l1 = l2 && List.length pred.(pc') = 1 ->
           remove_empty_jmp (pc+2)
-      | (Label _, _) when pred.(pc') = [pc-1] ->
+      | (Label _, _) when pred.(pc) = [pc-1] && succ (pc-1) = [pc] ->
```

Comment:
1. we need it because the single predecessor could be a branch:

     branch x l1 l2
    l1:

2. it does work but the reason is not obvious. The first check will be pred.(0) = [-1], therefore not fail but but certainly be false. therefore the second condition (which would fail at succ (-1)) will not be executed...